### PR TITLE
Export |ExclusiveAppComponent<Activity>| from FlutterActivity/FlutterFragment class

### PR DESCRIPTION
**Pull request by 0xZOne**
Rework based on review comments: https://github.com/flutter/engine/pull/30126#discussion_r765875835
Fixes: https://github.com/flutter/flutter/issues/93905

/cc @dnfield @blasten 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by dnfield — COMMENTED** [review]
This looks more reasonable to me, but I would defer to @blasten for review here.

File: shell/platform/android/io/flutter/embedding/android/FlutterActivity.java [hunk 1]
```diff
@@ -450,6 +450,14 @@ public FlutterActivity() {
     this.delegate = delegate;
   }
 
+  /**
+   * Returns the Android App Component exclusively attached to {@link
+   * io.flutter.embedding.engine.FlutterEngine}.
+   */
+  public ExclusiveAppComponent<Activity> getExclusiveAppComponent() {
```
Comment (blasten) [inline shell/platform/android/io/flutter/embedding/android/FlutterActivity.java:457]: nit: this should be defined in the interface. `FlutterActivityAndFragmentDelegate.Host`

**Review by blasten — APPROVED** [review]
small nit. Other than that LGTM

File: shell/platform/android/io/flutter/embedding/android/FlutterActivity.java (hunk 1 above)
Comment (0xZOne) [reply shell/platform/android/io/flutter/embedding/android/FlutterActivity.java:458]: Done.

File: shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java [hunk 2]
```diff
@@ -953,6 +953,12 @@ private void ensureAlive() {
     @NonNull
     TransparencyMode getTransparencyMode();
 
+    /**
+     * Returns the {@link ExclusiveAppComponent<Activity>} attached to {@link
```
Comment (blasten) [inline shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java:957]: this doesn't help much :/. How is this useful? why would anyone use this method?

Comment (0xZOne) [reply shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java]: Done.

File: shell/platform/android/io/flutter/embedding/android/FlutterActivity.java [hunk 3]
```diff
@@ -450,6 +450,23 @@ public FlutterActivity() {
...
+  /**
+   * Returns the {@link ExclusiveAppComponent<Activity>} that is associated with {@link
+   * io.flutter.embedding.engine.FlutterEngine}.
+   *
+   * <p>In the scenario where multiple {@link FlutterActivity} share the same {@link FlutterEngine},
+   * to attach/re-attache a {@link FlutterActivity} to the shared {@link FlutterEngine}, we MUST
+   * manually invoke {@link ActivityControlSurface#attachToActivity(ExclusiveAppComponent,
+   * Lifecycle)}.
+   *
+   * <p>The {@link ExclusiveAppComponent} is exposed here so that subclasses of {@link
```
Comment (blasten) [inline shell/platform/android/io/flutter/embedding/android/FlutterActivity.java:462]: nit: when this javadoc is specified in the interface, there's no need to copy it again in each implementation of the interface

File: shell/platform/android/io/flutter/embedding/android/FlutterActivity.java [hunk 4]
```diff
@@ -450,6 +450,23 @@ public FlutterActivity() {
     this.delegate = delegate;
   }
 
+  /**
+   * Returns the {@link ExclusiveAppComponent<Activity>} that is associated with {@link
```
Comment (blasten) [inline shell/platform/android/io/flutter/embedding/android/FlutterActivity.java:454]:
```
/b/s/w/ir/cache/builder/src/flutter/shell/platform/android/io/flutter/embedding/android/FlutterActivity.java:454: error: type arguments not allowed here
   * Returns the {@link ExclusiveAppComponent<Activity>} that is associated with {@link

```
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8824632744216046193/+/u/Build_and_test_arm64_profile/build_android_profile_arm64_default_clang_x64_gen_snapshot/stdout?format=raw

File: shell/platform/android/io/flutter/embedding/android/FlutterActivity.java (hunk 3 above)
Comment (0xZOne) [reply shell/platform/android/io/flutter/embedding/android/FlutterActivity.java:462]: thx, I got it.

File: shell/platform/android/io/flutter/embedding/android/FlutterActivity.java (hunk 4 above)
Comment (0xZOne) [reply shell/platform/android/io/flutter/embedding/android/FlutterActivity.java]: done.

**Review by blasten — APPROVED** [review]

**Review by dnfield — APPROVED** [review]

**Pull request merged**